tests/functional/CoreTest.php

```diff
@@ -40,23 +40,47 @@ public function testSiteSelectionViaCwd()
 
     public function testOptionsUri()
     {
-        // Put a yml file in the drush folder.
+        // Test whether a URI in a config file resolves correctly, and test
+        // various URI values for their expected Site URI and path.
         $drush_config_file = Path::join($this->webrootSlashDrush(), 'drush.yml');
-        $test_uri = 'http://test.uri';
-        $options_with_uri = [
-        'options' => [
-        'uri' => $test_uri,
-        ],
-        ];
-        $options = [
+        $command_options = [
         'format' => 'json',
         'uri' => 'OMIT', // A special value which causes --uri to not be specified.
         ];
-        file_put_contents($drush_config_file, Yaml::dump($options_with_uri, PHP_INT_MAX, 2));
-        $this->drush('core-status', [], $options);
-        unlink($drush_config_file);
-        $output = $this->getOutputFromJSON();
-        $this->assertEquals($test_uri, $output->uri);
+        // Modify settings.php to check whether site paths correctly resolve.
+        $root = $this->webroot();
+        $fp = fopen($root . '/sites/sites.php', 'a+');
+        fwrite($fp, '
+$sites["test.uri"] = "dev";
+$sites["test.uri.subpath"] = "stage";
+');
```

Comment:
I'm not sure if I should have reverted sites.php after this test finishes... You're not using "test.uri" anywhere else apparently, because tests still succeed.

Reply:
Shouldn't hurt to leave sites.php in place; you could even commit this to the git repository if you wanted. Optional.